remote-api/src/main/java/org/alfresco/rest/api/impl/NodesImpl.java

```diff
@@ -409,8 +409,7 @@ public NodeRef validateNode(StoreRef storeRef, String nodeId)
             nodeId = nodeId.substring(0, idx);
             if (versionLabel.equals("pwc"))
             {
-                // TODO correct exception?
-                throw new EntityNotFoundException(nodeId);
+                throw new InvalidArgumentException("Node with id: " + nodeId + "not found.");
```

Comment:
I'm usually in favour of fixing legacy code, but this exception (and TODO) has been here for over ten years!  Can we fix the issue with a slightly less scary change?  e.g. Catch the EntityNotFound and rethrow wrapped in an InvalidArgumentException?

I'd be interested to hear what others think about this too.

Reply:
I totally agree with you Tom! Changing ten years old exceptions which some logic may be dependent on is very risky.